compose.js

```diff
@@ -21,20 +60,29 @@ export const merge = (dst, src) => mergeWith(dst, src, (dstValue, srcValue) => {
  */
 function createFactory(descriptor) {
   return function Stamp(options = {}, ...args) {
-    let obj = Object.create(descriptor.methods || {});
+    // The 'methods' metadata object becomes the prototype for new object instances.
+    const obj = Object.create(descriptor.methods || {});
 
+    // Deep merge, then override with shallow merged properties, then apply property descriptors.
     merge(obj, descriptor.deepProperties);
     assign(obj, descriptor.properties);
     Object.defineProperties(obj, descriptor.propertyDescriptors || {});
 
+    // Run initializers sequentially.
     return (descriptor.initializers || [])
       .filter(isFunction)
       .reduce((resultingObj, initializer) => {
+        // Invoke an initializer in the way specification tell us to.
         const returnedValue = initializer.call(
+          // 'this' context will be the object instance itself
           resultingObj,
+          // the first argument passed from factory to initializer
           options,
+          // special arguments. See specification.
           {instance: resultingObj, stamp: Stamp, args: [options].concat(args)}
         );
+
+        // Any initializer can override the object instance with basically anything.
```

Comment:
Yes, babel is ES5 transpiler. Moreover, in this repo we do not about any older environment support because this is just a reference implementation of the `compose` function.